lib/nerves_time/real_time_clock.ex

```diff
@@ -0,0 +1,26 @@
+defmodule NervesTime.RealTimeClock do
+  @moduledoc """
+  Behaviour for hardware based real time clocks.
+  """
+
+  @typedoc "Internal state of the hardware clock"
+  @type state :: any()
+
+  @doc """
+  Initialize the hardware clock.
+  Called once when `ntpd` boots.
+  """
+  @callback init(args :: any()) :: {:ok, state} | {:error, reason :: any()}
+
+  @doc """
+  Retrieve the time from the hardware clock.
+  Called every time a stratum comes in from `ntpd`
+  """
+  @callback time(state) :: NaiveDateTime.t() | {:error, reason :: any()}
```

Comment:
Currently `FileTime.time/1` can't return error, but it does seem like the code to set a default time should be centralized instead of being in every rtc implementation. Could you change this to return `{:ok, NaiveDateTime.t()} | {:error, reason :: any()}` and update `FileTime.time/1` to return error if the file doesn't exist and move the default time to the code that calls this.

Also, in the function doc, this is only called on startup, right?

Reply:
ah you are correct, should only be called once on startup